orchestration/src/main/java/com/sap/ai/sdk/orchestration/LLMModuleResultDeserializer.java

```diff
@@ -0,0 +1,77 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.sap.ai.sdk.orchestration.client.model.LLMModuleResult;
+import com.sap.ai.sdk.orchestration.client.model.LLMModuleResultStreaming;
+import com.sap.ai.sdk.orchestration.client.model.LLMModuleResultSynchronous;
+import java.io.IOException;
+
+/**
+ * A deserializer for {@link LLMModuleResult} that determines the concrete implementation based on
+ * the structure of the JSON object.
+ */
+public class LLMModuleResultDeserializer extends StdDeserializer<LLMModuleResult> {
```

Comment:
(Question)

Do we need this class?

Reply:
> Do we need this class?

_"Deduction does not look recursively for discriminating field names. But that would be possible here."_

Reply:
No. We finalised to keep this class instead of `GeneralizedFallbackDeserializer` to handle oneOf for `LLMModuleResult`